internal/services/mongocluster/mongo_cluster_resource.go

```diff
@@ -440,13 +593,22 @@ func (r MongoClusterResource) Read() sdk.ResourceFunc {
 			if model := resp.Model; model != nil {
 				state.Location = location.Normalize(model.Location)
 
+				// Although Swagger defines four identity types, the service API currently only supports `None` and `UserAssigned`. Service team has confirmed that they will support the other types in the future.
+				identity, err := flattenMongoClusterIdentity(model.Identity)
+				if err != nil {
+					return err
+				}
```

Comment:
We should add consistent error handling per the contributor guidelines:
```suggestion
				if err != nil {
					fmt.Errorf("flattening identity: %+v", err)
				}
```
Reference: [reference-errors](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-errors.md)

Reply:
Updated